src/renderer/marketplace/keyfi/checkout/keyfi-checkout-container.jsx

```diff
@@ -0,0 +1,194 @@
+					this.cancelRoute(),
+					`KeyFi.ai Credentials`,
+					`You are about to begin the application process for getting credentials for KeyFi.ai.
+					Please double check your required documents are Certified True or Notarized where
+					necessary. Failure to do so will result in delays in the process. You may also be
+					asked to provide more information by the service provider`,
+					'conducting KYC',
+					vendor.name,
+					vendor.privacyPolicy,
+					vendor.termsOfService
+				)
+			);
+		}
+	};
+
+	onBackClick = () => this.props.dispatch(push(this.cancelRoute()));
+
+	// Status bar component allows for an action button on the right
+	// This handler processes the action for that button
+	onStatusActionClick = () => {
+		const { rp } = this.props;
+		if (rp && rp.authenticated && this.userHasApplied()) {
+			if (this.applicationCompleted() || this.applicationWasRejected()) {
+				this.props.dispatch(push(this.manageApplicationsRoute()));
+			} else if (this.applicationRequiresAdditionalDocuments()) {
+				this.redirectToKYCC(rp);
+			} else if (!this.userHasPaid()) {
+				this.props.dispatch(push(this.payRoute()));
+			} else if (!this.userHasSelectedBankPreference()) {
+				this.props.dispatch(push(this.selectBankRoute()));
+			}
+		}
+		return null;
+	};
+
+	render() {
+		const { cryptoCurrency } = this.state;
+		return (
+			<KeyFiCheckout
+				title={`Get your SelfKey Credentials to access KeyFi.ai`}
+				{...this.getPaymentParameters()}
+				price={this.props.product.price}
+				onBackClick={this.onBackClick}
+				onStartClick={this.onStartClick}
+				startButtonText={'Start Application'}
+				kycRequirements={this.props.kycRequirements}
+				templateId={this.props.templateId}
+				loading={this.props.loading}
+				cryptoCurrency={cryptoCurrency}
+				primaryToken={this.props.primaryToken}
+				onSelectCrypto={this.onSelectCrypto}
+				applicationStatus={this.getApplicationStatus()}
+				onStatusAction={this.onStatusActionClick}
+			/>
+		);
+	}
+}
+
+const mapStateToProps = (state, props) => {
+	const { vendorId, productId, templateId } = props.match.params;
+	const authenticated = true;
+	const primaryToken = { ...props, cryptoCurrency: config.constants.primaryToken };
+
+	const identity = identitySelectors.selectIdentity(state);
+	const product = marketplaceSelectors.selectInventoryItemBySku(
+		state,
+		'keyfi_kyc',
+		identity.type
+	);
+
+	const application = kycSelectors.selectApplications(state).find(app => app.rpName === 'keyfi');
+
+	return {
+		productId,
+		templateId,
+		vendorId,
+		product,
+		application,
+		vendor: marketplaceSelectors.selectVendorById(state, vendorId),
+		...getLocale(state),
+		...getFiatCurrency(state),
+		...ethGasStationInfoSelectors.getEthGasStationInfo(state),
+		tokens: getTokens(state).splice(1), // remove ETH
+		cryptoValue: getCryptoValue(state, primaryToken),
+		address: getWallet(state).address,
+		keyRate: pricesSelectors.getRate(state, 'KEY', 'USD'),
```

Comment:
will it work with KI?

Reply:
Some of those props were not needed. It should work with KI.
(Exchange rate is only available using KEY)